native/cocos/bindings/event/EventDispatcher.cpp

```diff
@@ -144,7 +143,8 @@ void EventDispatcher::dispatchTouchEvent(const TouchEvent &touchEvent) {
 
     se::ValueArray args;
     args.emplace_back(se::Value(jsTouchObjArray));
-    EventDispatcher::doDispatchEvent(nullptr, eventName, args);
+    EventDispatcher::doDispatchJsEvent(eventName, args);
+    EventDispatcher::doDispatchCustomEvent(nullptr, 0);
```

Comment:
Why separate doDispatchEvent into two functions?
If will be easier to forget one of the invocation.

And what's the purpose of passing `nullptr, 0` to doDispatchCustomEvent?

Reply:
The mechanism of the event dispatcher does not support the situation if we want to dispatch the native event with several arguments. for example:
```cpp
//Somewhere customize the render event
EventDispatcher::addCustomEventListener(EVENT_RESIZE, [&](CustomEvent e){
 texture.size.x = e.args[0].intVal;
 texture.size.y = e.args[1].intVal;
 windowHandle = e.args[2].ptrVal;
})
```

but doDispatchEvent will trigger Js events and Native events at same time, without adding arguments into the CustomEvent, so most of time, listeners can only be triggered with an event name.
```cpp
void EventDispatcher::doDispatchEvent(const char *eventName, const char *jsFunctionName, const std::vector<se::Value> &args) {
...
if (eventName) {
        CustomEvent event;
        event.name = eventName;
        EventDispatcher::dispatchCustomEvent(event);
    }
...
}
```
also, the listener will get errors if there are no args in the event.
That's why I want to separate it to 2 methods

Reply:
> And what's the purpose of passing nullptr, 0 to doDispatchCustomEvent?

I use va_list to accept multiple arguments, the disadvantage is that a descriptor should be given to specified the va_end